src/java/htsjdk/samtools/SAMUtils.java

```diff
@@ -945,14 +945,14 @@ public static int getMateUnclippedEnd(final SAMRecord rec) {
             } else {
                 if (getMateCigarString(rec) != null) {
                     ret = new ArrayList<SAMValidationError>();
-                    if (rec.getMateUnmappedFlag()) {
+                    if (!rec.getReadPairedFlag()) {
```

Comment:
can you explain what you worried about here? I think that this is equivalent due to the if statement above.